src/main/java/org/mitre/synthea/world/agents/behaviors/planeligibility/CSVEligibility.java

```diff
@@ -0,0 +1,118 @@
+    this.eligibilityCriteria = new ArrayList<>();
+    for (String key : inputEligibilities.keySet()) {
+      if (key.equals(LOGICAL_OPERATOR)) {
+        this.logicalOperator = convertToLogicalOperator(inputEligibilities.get(key));
+        continue;
+      }
+      if (key.equals(SUB_ELIGIBILITIES)) {
+        List<String> subEligibilities = Arrays.asList(inputEligibilities.get(key).split("\\|"));
+        subEligibilities.forEach(subEligibility -> this.eligibilityCriteria.add(
+            PlanEligibilityFinder.getEligibilityAlgorithm(subEligibility)));
+        continue;
+      }
+      if (!eligbilityOptions.containsKey(key)) {
+        throw new IllegalArgumentException("Invalid CSV eligibility input column: " + key + ".");
+      }
+      String input = inputEligibilities.get(key);
+      IPlanEligibility newEligbility = eligbilityOptions.get(key).apply(input);
+      this.eligibilityCriteria.add(newEligbility);
+    }
+  }
+
+  @Override
+  public boolean isPersonEligible(Person person, long time) {
+    if (logicalOperator.equalsIgnoreCase("AND")) {
+      return eligibilityCriteria.stream().allMatch(eligibility
+          -> eligibility.isPersonEligible(person, time));
+    }
+    if (logicalOperator.equalsIgnoreCase("OR")) {
+      return eligibilityCriteria.stream().anyMatch(eligibility
+          -> eligibility.isPersonEligible(person, time));
+    }
+    throw new RuntimeException("Error: invalid logical operator '"
+        + logicalOperator + "' for input eligibility.");
+  }
+
+  /**
+   * Builds the eligibility options and their mappings for all CSV input eligibilties.
+   */
+  static void buildEligibilityOptions(String state) {
+    eligbilityOptions = new HashMap<>();
+    eligbilityOptions.put(POVERTY_MULTIPLIER, (input)
+        -> new PovertyMultiplierEligibility(Double.parseDouble(input)));
+    eligbilityOptions.put(INCOME_THRESHOLD, (input)
+        -> new IncomeThresholdEligibility(Double.parseDouble(input)));
+    eligbilityOptions.put(AGE_THRESHOLD, (input)
+        -> new AgeThresholdEligibility(Integer.parseInt(input)));
+    eligbilityOptions.put(QUALIFYING_CONDITIONS, (input)
+        -> new QualifyingConditionCodesEligibility(input));
+    eligbilityOptions.put(QUALIFYING_ATTRIBUTES, (input)
+        -> new QualifyingAttributesEligibility(input));
+    eligbilityOptions.put(ACCEPTANCE_LIKELIHOOD, (input)
+        -> new AcceptanceLikelihoodEligibility(Double.parseDouble(input)));
+    eligbilityOptions.put(POVERTY_MULTIPLIER_FILE, (input)
+        -> new PovertyMultiplierFileEligibility(state, input));
+    eligbilityOptions.put(MNIL_FILE, (input)
+        -> new MedicallyNeedyIncomeEligibility(state, input));
+  }
+
+  /**
+   * Converts the given string to a logic operator.
+   * @param logicalOperator The input string.
+   * @return  The converted logical operator.
+   */
+  private static String convertToLogicalOperator(String logicalOperator) {
```

Comment:
Related to an above comment, you should probably trim whitespace on these to avoid throwing an exception that will be hard for people to understand. Essentially the error message would just have an extra whitespace in it.